src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -605,6 +606,27 @@ bool checkSymbol(Symbol sym, TextSpan memberSpan, SymbolKind kind, out Symbol re
                             }
                         }
                     }
+                    else if (sym.Kind == SymbolKind.Property)
+                    {
+                        if (InSpan(sym.GetFirstLocation(), this.syntaxTree, memberSpan))
+                        {
+                            return true;
+                        }
+
+                        // If this is a partial property, the property represents the defining part,
+                        // not the implementation (property.Locations includes both parts). If the
+                        // span is in fact in the implementation, return that property instead.
+                        var property = (SourcePropertySymbol)sym;
+                        var implementation = property.IsPartialDefinition ? property.OtherPartOfPartial : property;
+                        if ((object)implementation != null)
+                        {
+                            if (InSpan(implementation.GetFirstLocation(), this.syntaxTree, memberSpan))
```

Comment:
In the `!property.IsPartialDefinition` it looks like we're duplicating the ealier `InSpan` check. 
It may be better to introduce a `PartialImplementationPart` property like methods have.